content/en/docs/usage/gateway-api.md

```diff
@@ -0,0 +1,248 @@
+---
+title: "Securing Gateway Resources"
+linkTitle: "Securing Gateway Resources"
+weight: 90
+type: "docs"
+---
+
+Since 1.5, cert-manager supports requesting TLS certificates using annotations
+on Gateway resources. The Gateway resource is part of the [Gateway API][gwapi],
+a set of CRDs that can be installed to your Kubernetes cluster and that aim at
+offering an richer alternative to the Ingress abstraction.
+
+[gwapi]: https://gateway-api.sigs.k8s.io
+
+The Gateway resource holds the TLS configuration, as illustrated in the
+following diagram (source: https://gateway-api.sigs.k8s.io):
+
+![](https://gateway-api.sigs.k8s.io/images/gateway-roles.png)
+
+Note that cert-manager only supports setting up the TLS configuration on the
+Gateway resource when the Gateway is configured to terminate the TLS connection.
+There is still uncertainty around how the HTTPRoute TLS configuration should be
+used (see
+[gateway-api#577](https://github.com/kubernetes-sigs/gateway-api/issues/577)),
+and no existing implementation supports the TLS passthrough mode in which the
+HTTPRoute contains the TLS configuration and terminates the connection (see
+[istio#31747](https://github.com/istio/istio/issues/31747)).
+
+The Gateway support for Certificate shims must be enabled manually using a flag.
+If you are using Helm:
+
+```sh
+helm upgrade --install cert-manager jetstack/cert-manager
+  --set "extraArgs={--controllers=*\,gateway-shim}"
+```
+
+If you are using the raw cert-manager manifests, you can add the following like
+to the cert-manager controller Deployment `args`:
+
+```sh
+--controllers=*,gateway-shim
+```
+
+Like with an Ingress, the annotations `cert-manager.io/issuer` or
+`cert-manager.io/cluster-issuer` tells cert-manager that this Gateway should be
+looked at to create Certificate shims. For example, the following Gateway should
+trigger the creation of the `example-com-tls` Certificate:
+
+```yaml
+apiVersion: networking.x-k8s.io/v1alpha1
+kind: Gateway
+metadata:
+  name: example
+  annotations:
+    cert-manager.io/issuer: foo
+spec:
+  gatewayClassName: foo
+  listeners:
+    - hostname: example.com
+      port: 443
+      protocol: HTTPS
+      routes:
+        kind: HTTPRoute
+        selector:
+          matchLabels:
+            app: foo
+      tls:
+        mode: Terminate
+        certificateRef:
+          name: example-com-tls
+          kind: Secret
+          group: core
+```
+
+A few moments later, cert-manager should create a Certificate shim:
+
+```yaml
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: example-com-tls
+spec:
+  issuerRef:
+    name: my-issuer
+    kind: Issuer
+    group: cert-manager.io
+  dnsNames:
+    - example.com
+  secretName: example-com-tls
+```
+
+Contrary to Ingresses, you can set the same secret name multiple times. Let us
+imagine that you have these two listeners:
```

Comment:
I think these use case based descriptions are great! 
Perhaps worth to add a separate section 'Use cases' to this doc?
A few more I could think of 1) A Gateway with one 'valid' one 'invalid' listener block- a `Certificate` still gets created for the valid one. 2) A Gateway with two listeners each for different hostname and different secret name- two separate `Certificate`s get created for each. I guess eventually (but not for this PR?) it might make sense to describe this more from a user's perspective (i.e scenario based).

Reply:
Agreed, a "Use cases" section makes sense. I added it and had two subsections under it:

- "Skipping of invalid listener blocks"
- "Two listeners with the same Secret name"

> A Gateway with two listeners each for different hostname and different secret name- two separate Certificates get created for each.

I realized that you can also choose to re-use both the same hostname and secret name, I added an example with that.

> I guess eventually (but not for this PR?) it might make sense to describe this more from a user's perspective (i.e scenario based).

I struggle finding "actual" user-facing scenarios due to my lack of "actual" use of the Gateway API. As soon as I have more knowledge, I'd be happy to open a new PR with actual user scenarios. 👍